RedisEnterprise-DNS-Migration/migration_script.ps1

```diff
@@ -0,0 +1,325 @@
+        $name = $reCacheArmResource.Name;
+        $resourceGroup = $reCacheArmResource.ResourceGroupName;
+
+        $reCache = Get-AzRedisEnterpriseCache -Name $name -ResourceGroupName $resourceGroup;
+
+        $hostname = $reCache.HostName;
+
+        # Hostname will always be {section}.{region}.{suffix}, and this extracts just the suffix
+        $hostname -Match '^[^\.]+\.[^\.]+\.(.*)' | Out-Null
+        $suffix = $Matches[1]
+        $newZoneName = "privatelink.${suffix}"
+        Write-Host "Determined new zone name should be $newZoneName" -ForegroundColor Yellow
+        return $newZoneName
+    } catch {
+        # If we get here, the cache likely doesn't exist anymore and the private endpoint we're querying is in a disconnected state
+        Write-Host "The Cache $(($cacheResourceId -split "/")[-1]) doesn't exist. The private endpoint $($privateLink.Name) linked to it is in a $($serviceConnection.PrivateLinkServiceConnectionState.Status) connection state. Cannot set up private dns zones for this private endpoint. Please attach an existing cache." -ForegroundColor Red
+        $deletePrivateEndpoint = AskForConfirmation "Would you like to delete this private endpoint?"
+        if( $deletePrivateEndpoint ) {
+            $privateLink | Remove-AzPrivateEndpoint -Force | Out-Null
+        }
+        return $null
+    }
+}
+
+function TagAndRemoveLinksFromOldZones($vnetToZoneMap) {
+    Write-Host "Now that new zones are set up, the final step is to disconnect old zones. See step 'Testing your migration' in the migration document." -ForegroundColor Yellow 
+    Write-Host "As the old zones are disconnected they will be 'tagged' with old vnet they were attached to. This is to help with the 'Troubleshooting' section in the migration document, should it be needed." -ForegroundColor Yellow
+    foreach ($vnet in $vnetToZoneMap.GetEnumerator()) {
+        $zones = $vnetToZoneMap[$vnet.Name]
+        $oldZones = Get-PrivateDNSZonesMatchingPattern $OLD_DNS_ZONE_PATTERN $zones
+        if($null -eq $oldZones) {
+            Write-Host "No old zones exist in vnet $($vnet.Name)" -ForegroundColor Yellow
+        } else {
+            foreach($zone in $oldZones) {
+                
+                $links = Get-AzPrivateDnsVirtualNetworkLink -ResourceGroupName $zone.ResourceGroupName -ZoneName $zone.Name
+                foreach($link in $links) {
+                    if($link.VirtualNetworkId.ToLower() -eq $vnet.Name.toLower()) {
+                        $shouldContinue = AskForConfirmation "Deleting vnet link from $($zone.Name) to $($vnet.Name). Continue?"
+                        if ($shouldContinue) {
+                            Remove-AzPrivateDnsVirtualNetworkLink -ResourceGroupName $zone.ResourceGroupName -ZoneName $zone.Name -Name $link.Name | Out-Null
+                            Write-Host "Tagging old zone with information about the vnet is was linked to so it can be reconnected if needed" -ForegroundColor Yellow
```

Comment:
```diff
- Tagging old zone with information about the vnet is was linked to so it can be reconnected if needed
+ Tagging old zone with information about the vnet is was linked to. It can be reconnected if needed
```